packages/spindle-ui/src/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,170 @@
+    pageNumber: number,
+  ) => void;
+  createUrl: (pageNumber: number) => string;
+}
+
+const BLOCK_NAME = 'spui-Pagination';
+
+export const Pagination = (props: Props) => {
+  const {
+    current,
+    total,
+    showCount = false,
+    showPrevNext = true,
+    showFirstLast = false,
+    onPageChange,
+    createUrl,
+    className,
+    ...rest
+  } = props;
+
+  const pageItem = 5;
+  const { showItem, showPrevHorizontal, showNextHorizontal } = useShowItem({
+    current,
+    total,
+    pageItem,
+  });
+
+  const handleOnClick = useCallback(
+    (
+      event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+      pageNumber: number,
+    ) => {
+      onPageChange?.(event, pageNumber);
+    },
+    [onPageChange],
+  );
+
+  return (
+    <nav
+      aria-label="ページネーション"
+      className={[BLOCK_NAME, className].filter(Boolean).join(' ').trim()}
+      {...rest}
+    >
+      <ul className={`${BLOCK_NAME}-list`}>
+        {showFirstLast && (
+          <li
+            className={[`${BLOCK_NAME}-item`, `${BLOCK_NAME}-item--first`]
+              .join(' ')
+              .trim()}
+          >
+            <PaginationItem
+              type="first"
+              current={current}
+              total={total}
+              onClick={handleOnClick}
+              createUrl={createUrl}
+            />
+          </li>
+        )}
+        {showPrevNext && (
+          <li
+            className={[`${BLOCK_NAME}-item`, `${BLOCK_NAME}-item--prev`]
+              .join(' ')
+              .trim()}
+          >
+            <PaginationItem
+              type="prev"
+              current={current}
+              total={total}
+              onClick={handleOnClick}
+              createUrl={createUrl}
+            />
+          </li>
+        )}
+        {showItem.map((pageNumber, index) => {
+          return (
+            <li
+              className={`${BLOCK_NAME}-item`}
+              key={`pagination-item-${pageNumber}`}
+            >
+              {index === pageItem - 1 && showNextHorizontal && (
+                <MenuHorizontal
+                  aria-hidden="true"
+                  className={`${BLOCK_NAME}-horizontal`}
+                />
+              )}
+              <a
+                className={`${BLOCK_NAME}-link`}
+                href={
+                  current === pageNumber ? undefined : createUrl(pageNumber)
+                }
+                rel={current === pageNumber + 1 ? undefined : 'nofollow'}
+                aria-current={current === pageNumber ? 'page' : undefined}
+                aria-disabled={current === pageNumber ? true : undefined}
+                aria-label={`${pageNumber}ページ目`}
+                onClick={
+                  current === pageNumber
+                    ? undefined
+                    : (e) => {
+                        handleOnClick(e, pageNumber);
+                      }
+                }
+              >
+                {pageNumber}
+              </a>
```

Comment:
```suggestion
            {current === pageNumber ? (
                <a
                  className={`${BLOCK_NAME}-link`}
                  aria-label={`${pageNumber}ページ目`}
                  aria-current="page"
                  aria-disabled
                >
                  {pageNumber}
                </a>
              ) : (
                <a
                  className={`${BLOCK_NAME}-link`}
                  aria-label={`${pageNumber}ページ目`}
                  href={createUrl(pageNumber)}
                  rel="nofollow"
                  onClick={(e) => {
                    handleOnClick(e, pageNumber);
                  }}
                >
                  {pageNumber}
                </a>
              )}
```
nits. 同じ比較の分岐が多いため、マークアップごと分岐するかattributeのオブジェクトをuseMemoで返して使うようにするなどの方が可読性も含めて良いかもと思いました👀

Reply:
`current === pageNumber` と `current === pageNumber + 1` があって絶妙に分けづらいというのがありつつ。

https://github.com/openameba/spindle/pull/440/commits/81866bed7f093b0162248be1c8c3b0b8b01b9981#diff-0413a8895a7295898d7b0ae6eebfac7cec39b0cde6d8efcc06da3b85c2b36d43R82-R83

☝️変数にして気持ち見やすくなったかなーという感じなんですがどうですかねー。

Reply:
ありがとうございます！良さそうです！🙏